platform/aws.go

```diff
@@ -32,6 +32,12 @@ type awsMachine struct {
 	mach    *ec2.Instance
 }
 
+var _ Machine = (*awsMachine)(nil)
```

Comment:
Ohh, is this an assertion that it provides the interface? But doesn't NewMachine provide that assertion?